### ISSUE TITLE
Why did the FID score increase at the final layer in our experiment?

### ISSUE THREAD

**Issue by YeRookie** [shikiw/Modality-Integration-Rate#4]
![image](https://github.com/user-attachments/assets/c1b2ec77-e258-43c1-b52d-8738c3899778)

**Comment by shikiw** [comment]
<img width="580" alt="image" src="https://github.com/user-attachments/assets/afade3b6-af95-4cfa-97cb-7772546d4926">

The last layer hidden states gathered in llama's generate() will be processed by an additional RMSNorm layer, while other layer hidden states are the raw outputs of each LLM layer. You can exchange the order of the above two parts of code if you want to evaluate on the last layer hidden states without RMSNorm.

**Comment by YeRookie** [comment]
Could you help me analyze why do the MIR between visual features and textual features increases during alignment, even though the training loss is decreasing? It's worth mentioning that our training loss is decreasing. Although MIR increased slightly, the overall trend shows an upward trajectory.
<img width="320" alt="0a8da9cb707a9dcb1fea71b5d436e57" src="https://github.com/user-attachments/assets/68d615d4-20b1-448e-a313-9c8f08e119d6">

**Comment by YeRookie** [comment]
Could you help me analyze why do the MIR between visual features and textual features increases during alignment, even though the training loss is decreasing?  Although MIR increased slightly, the overall trend shows an upward trajectory. The FID of each layer is also gradually increasing during training.
<img width="320" alt="0a8da9cb707a9dcb1fea71b5d436e57" src="https://github.com/user-attachments/assets/043c308f-4985-4935-904a-bd8c83ea0525">
<img width="600" alt="b6dc09a7595381f082452cf046e39c3" src="https://github.com/user-attachments/assets/4215510d-3b24-4345-b6e2-cee4b0f40d9e">

**Comment by shikiw** [comment]
Hi, could you please provide more details about the model and data? Is this a pre-training stage? It seems to be the training data scale is far from to match the model scale.

**Comment by YeRookie** [comment]
Of course.Under the pre-trained 3B LLM, we aim to equip it with visual perception capabilities. Therefore, we trained an adapter from scratch to align the visual encoder with the LLM. In the figure, each checkpoint corresponds to a dataset size of 72k. Even when testing at steps below 500, MIR still shows a slow upward trend. It seems that MIR is not suitable for describing alignment and is more appropriate for SFT instead?

**Comment by shikiw** [comment]
MIR may not be entirely accurate in monitoring the initial steps of pre-training. In some cases, it might show an upward trend during the early stages, likely reflecting the warm-up period, where the model adjusts to a more optimization-friendly state. And in subsequent pre-training steps (with sufficient steps), MIR typically decreases and eventually converges. 

However, in your case, MIR continues to slightly increase during the entire pre-training, and I'm not sure if MIR still goes upward during your SFT. 

Here are some suggestions for further investigation: 

1. Carefully check if calculating MIR in a right way.
2. Adjust the training recipe and observe any resulting changes. 
3. Perform post-training on different pre-training checkpoints (e.g., checkpoint-500 and checkpoint-8000) to see if any differences on downstream benchmarks. 
4. If all above do not work, MIR may not be suitable as a monitoring tool for your pre-training. In this case, you can instead use it as a metric to evaluate fully pre-trained models or SFT models. 

Any feedback will be sincerely appreciated by us to further enhance the MIR designs :)

**Comment by YeRookie** [comment]
Thank you for your analysis. We have observed a decrease in MIR during a subsequent training. Thank you for this
excellent paper!
